Accessibility for Special Characters dialog

shoogle opened the pull request:
Fixes the F2 shortcut for the Insert Special Characters dialog and ensures the dialog is accessible to screen readers.

shoogle commented:
Tested on Windows and macOS. The F2 shortcut works on both platforms. Speech works with Narrator and NVDA. Speech for VoiceOver will be added when we port the dialog to QML: there's no easy way to do it before then unfortunately.

Eism commented:
@shoogle 
Crash occurs when pressing F2 without text editing
(you should add check in PaletteActionsController)

shoogle commented:
@Eism, thanks for the hint! I fixed that and added some more improvements. @Tantacrul has approved use of F2 as an alternative to Alt+Shift+E to begin text editing, and Shift+F2 to open the Special Characters dialog. F2 is the standard editing shortcut used in spreadsheet applications (to edit cell data) and File Explorer (to rename a file).

RobFog commented:
Rebase needed

shoogle commented:
@RobFog, I rebased the other day to account for the QString -> String change. It merges cleanly now.

DmitryArefiev commented:
Tested on Win10, Mac12, LinuxMint 20.2 

@shoogle 
1. F2 doesn't work for text editing for me (only Alt+Shift+E works)
2. As I see in the commits, Shift+F2 also should close the dialog but that doesn't work for me (added to #11145 for now)

shoogle commented:
@DmitryArefiev
> F2 doesn't work for text editing for me (only Alt+Shift+E works)

Strange! F2 and Alt+Shift+E are both working for me on all platforms. If you're using a laptop I guess you already tried pressing Fn with F2? Maybe Help > Revert to Factory Settings and relaunching MuseScore will help? Or maybe there is another program stealing the F2 keypress event (e.g. a screen reader or virtual machine app)?

> Shift+F2 also should close the dialog but that doesn't work for me

It works when the main window has focus (i.e. when the Insert Special Characters dialog doesn't have focus). When the dialog has focus you are able to close it with Esc. I can try to enable Shift+F2 to close the dialog even when it has focus, but it wouldn't work if you manually change the Shift+F2 shortcut to a different key combination in the Preferences (at least not without an overhaul of the shortcuts system).

Eism commented:
> When the dialog has focus you are able to close it with Esc. I could try to enable Shift+F2 to close the dialog even when it has focus, but I think that's a separate issue really.

I think Esc is enough

Eism commented on src/engraving/libmscore/textedit.cpp:
```diff
@@ -249,9 +249,14 @@ bool TextBase::isEditAllowed(EditData& ed) const
     bool shiftPressed = ed.modifiers & Qt::ShiftModifier;
 
+    if (shiftPressed && ed.key == Qt::Key_F2) {
+        return false; // Toggle Special Characters dialog
```
I think we can find a better solution
see https://github.com/musescore/MuseScore/pull/11923#issuecomment-1166914602

Eism replied:
Okey, let's leave it as it is for now. Many changes will be needed

DmitryArefiev commented:
> Strange! F2 and Alt+Shift+E are both working for me on all platforms. If you're using a laptop I guess you already tried pressing Fn with F2? Maybe Help > Revert to Factory Settings and relaunching MuseScore will help? Or maybe there is another program stealing the F2 keypress event (e.g. a screen reader or virtual machine app)?

@shoogle Yes, after Revert to Factory Settings applying F2 works fine. Thanks!

> It works when the main window has focus (i.e. when the Insert Special Characters dialog doesn't have focus). When the dialog has focus you are able to close it with Esc. I could try to enable Shift+F2 to close the dialog even when it has focus, but I think that's a separate issue really. It should be easy enough to handle the Shift+F2 keypress when the dialog has focus but it would require more effort to get it working if the user has reassigned the shortcut to a different key combination.

Yes, let's track it in #11145 and it's not major because Esc works fine

shoogle commented:
Ok cool! I guess this is ready to merge then?
